evm/generators.go

```diff
@@ -1281,131 +1598,193 @@ func {{.HandlerName}}() *cobra.Command {
+    var safeAddress, safeApi string
+	var safeOperation uint8
+
+    {{range .MethodArgs}}
+    var {{.CLIVar}} {{.CLIType}}
+    {{if (ne .CLIRawVar .CLIVar)}}var {{.CLIRawVar}} {{.CLIRawType}}{{end}}
+    {{- end}}
+
+    cmd := &cobra.Command{
+        Use: "{{(KebabCase .MethodName)}}",
+        Short: "Execute the {{.MethodName}} method on a {{$structName}} contract",
+        PreRunE: func(cmd *cobra.Command, args []string) error {
+            if contractAddressRaw == "" {
+                return fmt.Errorf("--contract not specified")
+            } else if !common.IsHexAddress(contractAddressRaw) {
+                return fmt.Errorf("--contract is not a valid Ethereum address")
+            }
+            contractAddress = common.HexToAddress(contractAddressRaw)
+
+            if keyfile == "" {
+                return fmt.Errorf("--keystore not specified (this should be a path to an Ethereum account keystore file)")
+            }
+
+			if rpc == "" {
+				return fmt.Errorf("--rpc not specified (this should be a URL to an Ethereum JSONRPC API)")
 			}
 
-			chainIDCtx, cancelChainIDCtx := NewChainContext(timeout)
-			defer cancelChainIDCtx()
-			chainID, chainIDErr := client.ChainID(chainIDCtx)
-			if chainIDErr != nil {
-				return chainIDErr
-			}
-
-			transactionOpts, transactionOptsErr := bind.NewKeyedTransactorWithChainID(key.PrivateKey, chainID)
-			if transactionOptsErr != nil {
-				return transactionOptsErr
-			}
-
-			SetTransactionParametersFromArgs(transactionOpts, nonce, value, gasPrice, maxFeePerGas, maxPriorityFeePerGas, gasLimit, simulate)
-
-			contract, contractErr := New{{$structName}}(contractAddress, client)
-			if contractErr != nil {
-				return contractErr
-			}
-
-			session := {{$structName}}TransactorSession{
-				Contract: &contract.{{$structName}}Transactor,
-				TransactOpts: *transactionOpts,
-			}
-
-			transaction, transactionErr := session.{{.MethodName}}(
-				{{- range .MethodArgs}}
-				{{.CLIVar}},
-				{{- end}}
-			)
-			if transactionErr != nil {
-				return transactionErr
-			}
-
-			cmd.Printf("Transaction hash: %s\n", transaction.Hash().Hex())
-			if transactionOpts.NoSend {
-				estimationMessage := ethereum.CallMsg{
-					From: 		transactionOpts.From,
-					To: 		&contractAddress,
-					Data: 		transaction.Data(),
+            if safeAddress != "" {
+                if !common.IsHexAddress(safeAddress) {
+                    return fmt.Errorf("--safe is not a valid Ethereum address")
+                }
+                if safeApi == "" {
+					client, clientErr := NewClient(rpc)
+					if clientErr != nil {
+						return clientErr
+					}
+					chainIDCtx, cancelChainIDCtx := NewChainContext(timeout)
+					defer cancelChainIDCtx()
+					chainID, chainIDErr := client.ChainID(chainIDCtx)
+					if chainIDErr != nil {
+						return chainIDErr
+					}
+					safeApi = "https://safe-client.safe.global/v1/chains/" + chainID.String() + "/transactions/" + safeAddress + "/propose"
```

Comment:
I prefer `fmt.Sprintf` instead of constructing the string like this. Just a nit,  though.

Reply:
Excellent, I will implement it so we keep the pattern